T2522 implement user settings

Nico812 opened the pull request:
[Screencast from 12.08.2025 11:38:21.webm](https://github.com/user-attachments/assets/1dfea387-1bd5-4f0d-b554-f287f4d7f8ee)

The change password button sends to an old password change page.

Some fields are updated in the database without checking if they are valid. For example the phone number can be a random string. The only check that I do here is that the fields are not empty.

For the email there was already a check in the backend (maybe an odoo default one, or maybe something implemented by Emanuel) and i caught that one for displaying if an email is valid. This can be seen in the screen recording.

For the autentication email changement, i check that the email doesn't already exists in the database, as shown in the screen recording.

Commit:
[T2370] Just added the new page, nothing interesting. The commit is so that i can switch branch to take some code from daniel

Signed-off-by: nico <nhepp@compassion.ch>

Commit:
[T2370] Just added the new page, nothing interesting. The commit is so that i can switch branch to take some code from daniel

Signed-off-by: nico <nhepp@compassion.ch>

Commit:
[T2370] Started implementing the logic for the option to change the personal informations.

Signed-off-by: nico <nhepp@compassion.ch>

Commit:
[T2370] Personal informations now can be changed. Started working on the other tabs

Signed-off-by: nico <nhepp@compassion.ch>

Commit:
[T2370] Committing before refactoring the communication preferences using a new technique Nathan showed me

Signed-off-by: nico <nhepp@compassion.ch>

Commit:
[T2370] Seems stable, still need to change some buttons style

Signed-off-by: nico <nhepp@compassion.ch>

Commit:
[T2522] Almost finished, everything works except "edit account settings" button and still need to change the style of a tab selector

Signed-off-by: nico <nhepp@compassion.ch>

Commit:
[T2522] Committing before a refactor so in case I can come back

Signed-off-by: nico <nhepp@compassion.ch>

Commit:
[T2522] Almost done, committing to take some code from nathan's branch

Signed-off-by: nico <nhepp@compassion.ch>

Commit:
[T2522] Switching branch

Signed-off-by: nico <nhepp@compassion.ch>

Commit:
[T2522] Good state. Committing before rebase.

Signed-off-by: nico <nhepp@compassion.ch>

Commit:
[T2522] Good state. Committing before rebase.

Signed-off-by: nico <nhepp@compassion.ch>

Nico812 committed:
Merge branch '14.0-MyCompassion2.0' into T2522-implement-user-settings

Nico812 commented on my_compassion/controllers/my2_user_settings.py:
```diff
@@ -0,0 +1,131 @@
+        allowed_fields = {
+            'title': int, 'lastname': str, 'firstname': str, 'street': str,
+            'city': str, 'country_id': int, 'zip': str, 'phone': str, 'email': str
+        }
```
let me know if i have to address this. Because I'm not sure it's part of the task

NoeBerdoz replied:
This issue will be addressed in another task.

Commit:
[T2522] Addressed some minor changes

Signed-off-by: nico <nhepp@compassion.ch>

Nico812 commented on my_compassion/static/src/js/my2_user_settings.js:
```diff
@@ -0,0 +1,261 @@
+         * Generic form handler for tabs with an "edit/save/cancel" workflow.
+         */
+        function initFormHandler({ formId, editButtonId, saveButtonId, cancelButtonId, endpoint, fields }) {
+            console.log("CACCA")
```
already fixed it

Nico812 commented on my_compassion/templates/pages/my2_user_settings.xml:
```diff
@@ -0,0 +1,507 @@
+                                               for="flexCheckDefault">
+                                            I accept the legal terms and the privacy
+                                            policy conditions
+                                        </label>
```
already fixed it

Commit:
[T2522] pre-commit changes

Signed-off-by: nico <nhepp@compassion.ch>

Commit:
[T2522] pre-commit changes

Signed-off-by: nico <nhepp@compassion.ch>

danpa32 commented:
<img width="965" height="607" alt="Screenshot" src="https://github.com/user-attachments/assets/d7ae1649-f84d-491f-87c5-60559ae6a180" />

Fixing for better UI/UX:

- Let's put First name and Lastname and reversed them (first name first, lastname after)
- Putting the City then the Zip then the address is a bit weird, so let's do 1 line the country, 1 line the Zip and City, 1 line the Address and number

danpa32 commented:
<img width="836" height="487" alt="image" src="https://github.com/user-attachments/assets/0f2c8824-bc5e-4f90-a286-447ceaa073ac" />

<img width="738" height="345" alt="Screenshot" src="https://github.com/user-attachments/assets/44a8d60a-882c-408d-96da-13b837acddb7" />

When you hover the mouse on this buttons, there is an underline that comes. We have to remove it

danpa32 commented:
<img width="941" height="479" alt="2025-08-15_12-51" src="https://github.com/user-attachments/assets/f8869284-ca59-4a01-98a8-6ae1268ff0f0" />

https://www.figma.com/design/JfJA1ufhQREDJXr7m7FxSI/MyCompassion-2.0?node-id=1465-5544&t=4mCx8v4dQL6xBOZx-0

Between the implementation and the design, there is disparity:
- The order of elements should be reversed
- The link is missing for the data protection (https://compassion.ch/protection-des-donnees/)

We have to add a signature function (button) in the child protection charter like it is currently in mycompassion.ch 


For better readability, we have to have 2 paragraph for the text in the blue rectangle. We should have

> 
> Compassion knows that children living in extreme poverty are vulnerable. We put their protection as our number one priority. We ask our volunteering friends and people that travel to visit their sponsored child to sign a child protection charter. If you want to make a step further, you can sign this document as well and show us that you share our values. 
> 
> Compassion is also serious about the protection of your data. All your information is kept private and only used internally. You can sign or review our data protection policy from here.


<img width="1866" height="916" alt="2025-08-15_13-02" src="https://github.com/user-attachments/assets/a7a30900-224a-408f-a52f-5f4d2782335c" />

- As the protection charter page for the child, it should not take the whole size of the screen but 768px is good for readability. There is a scss class named _content-column_ for that in the theme_compassion_2025. For reminder in scss theme class, _site-wrapper_ makes the content 1920px width (good to encapsulate the whole content page), _container_ = 1200px width (good for smaller screen, tablet and some) and _content-column_ = 768px width (good for reading text and not having to much word in a long line)

NoeBerdoz commented on my_compassion/static/src/js/my2_user_settings.js:
```diff
@@ -0,0 +1,268 @@
+         * Initializes the child protection charter agreement checkbox.
+         */
+        function initPrivacyForm() {
+            const checkbox = document.getElementById("flexCheckDefault");
```
'flexCheckDefault' id doesn't exist on the DOM side, the checkbox is not working correctly.

NoeBerdoz reviewed (changes requested):
Thank you for your work, changes are needed to be done, please refer to @danpa32's comments and make sure to fix the Child Protection related functionality, it doesn't work as intended

Commit:
[T2522] Fixed all the problems raised in the PR

Signed-off-by: nico <nhepp@compassion.ch>

Nico812 commented:
[Screencast from 19.08.2025 14:41:04.webm](https://github.com/user-attachments/assets/43b84299-fee9-40a4-b766-65223e8e84a2)

Thats how the new privacy settings tab looks

Commit:
[T2522] Precommit

Signed-off-by: nico <nhepp@compassion.ch>

Nico812 committed:
Merge branch '14.0-MyCompassion2.0' into T2522-implement-user-settings

Nico812 commented:
The precommit errors are not in the files i changed in my branch

NoeBerdoz commented on my_compassion/templates/pages/my2_user_settings.xml:
```diff
@@ -0,0 +1,624 @@
+                                        <t t-set="label" t-value="'Phone number'" />
+                                        <t t-set="input_id" t-value="'phone'" />
+                                        <div class="field-display" data-display-for="phone">
+                                            <t t-esc="partner.phone or ''" />
+                                        </div>
+                                        <div class="field-edit">
+                                            <input
+                                                type="tel"
+                                                id="phone"
+                                                name="phone"
+                                                class="form-control"
+                                                t-att-value="partner.phone"
+                                                t-att-data-original-value="partner.phone"
+                                            />
+                                        </div>
```
Partner can either have a phone, or a mobile phone, it wasn't present on the figma design, but both should be handled

NoeBerdoz commented on my_compassion/templates/pages/my2_user_settings.xml:
The Address part should follow this structure:
Address, Zip, City, Country.

Here is a proposition:
<img width="717" height="581" alt="image" src="https://github.com/user-attachments/assets/b293d5a1-5816-4164-9d44-03111d878af5" />

NoeBerdoz reviewed (changes requested):
Thanks for the corrections @Nico812, there is still a few things to do before we can merge this PR:
- Redo the address related fields positions
- Mobile phone handling
- Updating the tax delivery preference doesn't work properly in my testings, it needs to be fixed

Commit:
Fixed the account settings tab, still waiting to fix user tax receipt preference

Signed-off-by: nico <nhepp@compassion.ch>

Commit:
Fixed the issues brought up in the PR

Signed-off-by: nico <nhepp@compassion.ch>

Nico812 commented:
<img width="789" height="846" alt="Screenshot from 2025-08-25 13-02-07" src="https://github.com/user-attachments/assets/71b4de4c-2e49-4718-bcc9-983f22ac1763" />

fixed this layout and addressed all the things brougth up in this PR

NoeBerdoz committed:
Merge branch '14.0-MyCompassion2.0' into T2522-implement-user-settings

# Conflicts:
#	theme_compassion_2025/static/src/scss/components/_form_field.scss
#	theme_compassion_2025/templates/components/FormField.xml

NoeBerdoz committed:
add missing newline at end of file

NoeBerdoz commented:
pre-commit is failing for code not related to the PR

NoeBerdoz reviewed (approved):
Good job @Nico812, thanks for your corrections
